src/index_runner/event_loop.py

```diff
@@ -0,0 +1,123 @@
+'''
+The event loop for the index runner.
+'''
+
+import json
+import logging
+import os
+import requests
+import time
+import traceback
+
+from typing import Callable, Dict, Any
+
+from confluent_kafka import Consumer, KafkaError
+
+from src.utils.config import config
+from src.utils.service_utils import wait_for_dependencies
+
+Message = Dict[str, Any]
+
+# TODO TEST unit tests
+
+
+def start_loop(
+        consumer: Consumer,
+        message_handler: Callable[[Message], None],
+        on_success: Callable[[Message], None] = lambda msg: None,
+        on_failure: Callable[[Message, Exception], None] = lambda msg, e: None,
+        on_ready: Callable[[], None] = lambda: None,
+        on_config_update: Callable[[], None] = lambda: None,
+        logger: logging.Logger = logging.getLogger('src.event_loop')):
```

Comment:
My experience has been the logger name = the module / class name. I don't have strong opinion here though, so I'll change it to IR

Reply:
Actually, I think passing as a param is better since then you can customize the logger however you want (or even pass in a completely different class with the same interface) as pass that in. If the logger is created as a global then you can't do that without monkey patching.

Reply:
Changed name to IR